style_guide.md

```diff
@@ -0,0 +1,304 @@
+    // Imports
+    
+    // Variant helper methods
+    local questionTitle(variantParam) = (
+      // Variables relevant to this method
+      // Return relevant string or JSON object
+    );
+    
+    // Main variant method - 'question' or 'content'
+    local question(variantParam) = {
+      // Variant JSON
+    };
+    
+    {
+      // Block JSON
+    }
+    ```
+### Block json
+
+- Block json can be a function if an argument is being passed:
+
+    ```
+    function(region_code, census_month_year_date) {
+      type: 'Question',
+      id: 'arrive-in-country',
+      question_variants: [
+        {
+          question: question(nonProxyTitle),
+          when: [rules.isNotProxy],
+        },
+        {
+          question: question(proxyTitle),
+          when: [rules.isProxy],
+        },
+      ],
+    } 
+    ```
+  
+### Variants
+
+- When the value of a property needs to vary, a method should be defined that returns an appropriate value.
+
+#### Single value variants example
+
+- Most common approach in census schemas is to create variants based on a value of a single variable.
+- Example of this would be a `date of birth` question using `isProxy` variants:
+    ```
+    {
+      type: 'Question',
+      id: 'confirm_dob',
+      question_variants: [
+        {
+          question: question(isProxy=false),
+          when: [rules.isNotProxy],
+        },
+        {
+          question: question(isProxy=true),
+          when: [rules.isProxy],
+        },
+      ],
+    }
+    ```
+- `Question` method passes `isProxy` variable to `local question`:
+    ```
+    local question(isProxy) = {
+      title: questionTitle(isProxy)
+      id: 'confirm-date-of-birth',
+      title: title,
+      type: 'General',
+      answers: [
+      {
+        id: 'confirm-date-of-birth-answer',
+        mandatory: true,
+        options: [
+          {
+            label: 'Yes',
+            value: 'Yes',
+           },
+          {
+            label: 'No',
+            value: 'No',
+          },
+        ],
+        type: 'Radio',
+        },
+      ],
+    };
+    ```
+- In the `local question`, when we need to resolve title, we call a new `questionTitle` method, again using `isProxy`:
+    ```
+    local questionTitle(isProxy) = (
+      if isProxy == true then {
+      text: 'What was <em>{person_name_possessive}</em> age on their last birthday?',
+      placeholders: [
+        placeholders.personNamePossessive,
+      ]} 
+      else 'What was your age on your last birthday?';
+    );
+    ```
+- We would return either of the two strings, based on the `isProxy` variable.
+- Same rules apply to other blocks with variants, e.g. for guidance with a `questionGuidance` method we would use:
+    ```
+    local questionGuidance(isProxy) = (
+      if isProxy == true then 'Why we ask this question if they are retired or long-term sick or disabled' else 'Why we ask this question if you are retired or long-term sick or disabled';
+    );
+    ```
```

Comment:
I think it would be clearer if the full example was in one code block. It will also make it clear what ordering exists in the Jsonnet.

- The most common approach in census schemas is to create variants based on a value of a single variable:
```
Full code block
```

- The `isProxy` variable is passed to the `question` method
- The `question` method calls `questionTitle(isProxy)` to resolve the question title
- `questionTitle` returns the appropriate JSON for title

Reply:
I used your bullet points and moved it to single block, thanks!